engine/src/main/java/com/datatorrent/stram/client/AppPackage.java

```diff
@@ -329,6 +330,13 @@ public void processAppDirectory(boolean skipJars)
 
     Configuration config = new Configuration();
 
+    // load configuration with default properties
+    if (MapUtils.isNotEmpty(defaultProperties)) {
+      for (String key : defaultProperties.keySet()) {
```

Comment:
@sandeshh updated code as per comments . Will add unit tests in couple of days and will inform you to review tests at that time.